types/chai/index.d.ts

```diff
@@ -1210,7 +1210,11 @@ declare namespace Chai {
          * @param length   Potential expected length of object.
          * @param message   Message to display on error.
          */
-        lengthOf<T extends { readonly length?: number | undefined }>(object: T, length: number, message?: string): void;
+        lengthOf<T extends { readonly length?: number | undefined } | { readonly size?: number | undefined }>(
```

Comment:
This isn't a new problem, but I don't understand why this is a generic at all. It could just accept this constraint as the parameter type.

That and, this type technically accepts the literal `{}` as an argument, which feels wrong.